contract_api/registry.py

```diff
@@ -0,0 +1,274 @@
+import datetime
+import json
+from collections import defaultdict
+from common.utils import Utils
+from pymysql import MySQLError
+from contract_api.filter import Filter
+
+
+class Registry:
+    def __init__(self, obj_repo):
+        self.repo = obj_repo
+        self.obj_utils = Utils()
+
+    def _get_all_service(self):
+        """ Method to generate org_id and service mapping."""
+        try:
+            all_orgs_srvcs_raw = self.repo.execute("SELECT O.org_id, O.owner_address, S.service_id  FROM service S, "
+                                                   "organization O WHERE S.org_id = O.org_id AND S.is_curated = 1")
+            all_orgs_srvcs = {}
+            for rec in all_orgs_srvcs_raw:
+                if rec['org_id'] not in all_orgs_srvcs.keys():
+                    all_orgs_srvcs[rec['org_id']] = {'service_id': [],
+                                                     'owner_address': rec['owner_address']}
+                all_orgs_srvcs[rec['org_id']]['service_id'].append(
+                    rec['service_id'])
+            return all_orgs_srvcs
+        except Exception as e:
+            print(repr(e))
+            raise e
+
+    def _get_all_members(self):
+        """ Method to generate org_id and members mapping."""
+        try:
+            all_orgs_members_raw = self.repo.execute(
+                "SELECT org_id, member FROM members M")
+            all_orgs_members = defaultdict(list)
+            for rec in all_orgs_members_raw:
+                all_orgs_members[rec['org_id']].append(rec['member'])
+            return all_orgs_members
+        except Exception as e:
+            print(repr(e))
+            raise e
+
+    def get_all_org(self):
+        """ Method to get list and high level details of all organization."""
+        try:
+            all_orgs_srvcs = self._get_all_service()
+            all_orgs_members = self._get_all_members()
+            all_orgs_data = []
+            for rec in all_orgs_srvcs:
+                data = {"org_id": rec,
+                        "owner_address": all_orgs_srvcs[rec]['owner_address'],
+                        "service_id": all_orgs_srvcs[rec]['service_id'],
+                        "members": all_orgs_members.get(rec, [])}
+                all_orgs_data.append(data)
+            return all_orgs_data
+        except Exception as e:
+            print(repr(e))
+            raise e
+
+    def _prepare_subquery(self, s, q, fm):
+        try:
+            sub_qry = ""
+            if s == "all":
+                for rec in fm:
+                    sub_qry += fm[rec] + " LIKE '" + str(q) + "%' OR "
+                sub_qry = sub_qry[:-3] if sub_qry.endswith("OR ") else sub_qry
```

Comment:
We should not be doing `sub_qry[:-3].` Lets explicitly add an OR if sub_qry is not empty. Something like

`for rec in fm:
     if len(sub_qry) > 0:
             sub_qry += ' OR '
     <logic here>
`